src/reducers/ui-state.js

```diff
@@ -36,9 +36,7 @@ const actionHandler = {
   [ActionTypes.ADD_NOTIFICATION]: uiStateUpdaters.addNotificationUpdater,
   [ActionTypes.REMOVE_NOTIFICATION]: uiStateUpdaters.removeNotificationUpdater,
 
-  [ActionTypes.SET_RATIO]: uiStateUpdaters.setRatioUpdater,
-  [ActionTypes.SET_RESOLUTION]: uiStateUpdaters.setResolutionUpdater,
-  [ActionTypes.TOGGLE_LEGEND]: uiStateUpdaters.toggleLegendUpdater,
```

Comment:
why removing toggle legend?

Reply:
because all these 3 actions are behaving the same and will trigger recalculation of imageSize. they are merged into 1 below